### PR TITLE
Add "display post date" and "number of posts to show" Inspector options for Latest Posts block

### PR DESCRIPTION
**Pull request by lamosty**
When [building the Latest Posts block](https://github.com/WordPress/gutenberg/pull/870), @jasmussen did [a great design](https://github.com/WordPress/gutenberg/pull/870#issuecomment-303326566) of some initial Inspector options. In this PR, we are implementing parts of it.

## Testing
1. In a post, add the latest posts block and click on it;
2. On the left of the screen, the Gutenberg Inspector should show the "display post date" and "number of posts to show" options;
3. Play with the options -- do they work as expected?

Looking like this as of now:

selection_359

## TODO
- [ ] Add display post date to server-side rendering
- [ ] Possibly debounce the change in number of posts to show input

### CODE REVIEW

**Comment by westonruter** [comment]
@lamosty I was doing this as well in #1183.

**Comment by westonruter** [comment]
Rebased to remove merge commit 13bcbf6 (and former `HEAD`).

File: blocks/library/latest-posts/index.js [hunk 1]
```diff
@@ -50,23 +91,62 @@ registerBlockType( 'core/latestposts', {
...
+							<label htmlFor={ postToShowId }>{ __( 'Number of posts to show' ) } </label>
+							<input
+								type="text"
+								value={ this.props.attributes.poststoshow }
+								ref={ postToShowId }
+								id={ postToShowId }
+								className="editor-latest-posts__input"
+								onChange={ () => this.changePostsToShow( this.refs[ postToShowId ].value ) }
+							/>
+						</div>
```
Comment (jasmussen) [inline blocks/library/latest-posts/index.js]: We should absolutely use the same controls everywhere, but whether this happens post merge or not I don't think is important. Should I open a separate ticket for making sure controls are all the same?

**Comment by westonruter** [comment]
So now this is waiting on #1278?

**Comment by jasmussen** [comment]
> So now this is waiting on #1278?

I don't think it has to. IMO okay to merge so long as we remember to address issues after the fact.

**Comment by jasmussen** [comment]
Took the liberty of pushing a tiny style tweak to the post date:

screen shot 2017-06-23 at 11 57 01

File: blocks/library/latest-posts/index.js [hunk 2]
```diff
@@ -35,6 +40,42 @@ registerBlockType( 'core/latestposts', {
...
+
+			setAttributes( { displayPostDate: ! displayPostDate } );
+		}
+
+		componentWillReceiveProps( nextProps ) {
+			const { poststoshow: postToShowCurrent } = this.props.attributes;
+			let { poststoshow: postToShowNext } = nextProps.attributes;
+			const { setAttributes } = this.props;
+
+			postToShowNext = parseInt( postToShowNext );
```
Comment (youknowriad) [inline blocks/library/latest-posts/index.js]: The `parseInt` is a bit weird, we should get the JSON attributes PR merged soon :)

File: blocks/library/latest-posts/index.js (hunk 1 above)
Comment (youknowriad) [inline blocks/library/latest-posts/index.js]: The controls are merged, let's rebase this :)

File: blocks/library/latest-posts/index.js [hunk 3]
```diff
@@ -19,11 +22,13 @@ registerBlockType( 'core/latestposts', {
 
 	defaultAttributes: {
 		poststoshow: 5,
+		displayPostDate: false,
 	},
 
-	edit: class extends wp.element.Component {
+	edit: withInstanceId( class extends wp.element.Component {
```
Comment (youknowriad) [inline blocks/library/latest-posts/index.js]: We're moving way from `wp.*` as much as we can in favor of `import { Component } from 'element';`

File: blocks/library/latest-posts/index.js (hunk 2 above)
Comment (iseulde) [inline blocks/library/latest-posts/index.js]: Please always add the `radix`. :)

Comment (youknowriad) [inline blocks/library/latest-posts/index.js]: I think we should probably drop it, since the JSON attributes are merged.

**Comment by iseulde** [comment]
Rebasing...

**Comment by iseulde** [comment]
What else is left here?

**Comment by jasmussen** [comment]
I think it's good to go, probably!

**Comment by iseulde** [comment]
@jasmussen Fixed padding issue that I missed. Should be good to go for now.

There's a rule that's overly broad in Gutenberg:

```scss
.gutenberg ul,
.gutenberg ol {
  margin: 0;
  padding: 0;
}
```

I'm not sure why this is done, because now blocks are forced to overwrite it by adding a parent selector.

**Comment by iseulde** [comment]
Also adjusted the value: `padding-left: 2.5em;`. This should probably be set by the editor, not the block, so it is inherited by both the list block and the latest post block.

**Comment by jasmussen** [comment]
You're right, we should probably scope that UL code a bit. I can't recall the circumstances right now, but I think it's just basic CSS normalization.

**Comment by iseulde** [comment]
Okay, let's merge and iterate.

**Comment by lamosty** [comment]
Thanks for finishing this, I couldn't find time for it, sorry!

**Pull request merged**